src/app-layout/visual-refresh/drawers.tsx

```diff
@@ -396,21 +358,31 @@ export function MobileTriggers() {
           badgeColor="red"
           key={item.id}
           onClick={() => handleDrawersClick(item.id)}
-          ref={item.id === previousActiveDrawerId.current ? drawersRefs.toggle : undefined}
           variant="icon"
           __nativeAttributes={{ 'aria-haspopup': true, 'data-testid': `awsui-app-layout-trigger-${item.id}` }}
         />
       ))}
-      {overflowItems && overflowItems.length > 0 && (
+      {overflowItems.length > 0 && (
         <OverflowMenu
-          drawersRefs={drawersRefs}
-          className={clsx(styles['drawers-trigger'], overflowItemHasBadge && styles.badge)}
-          overflowItems={overflowItems}
+          items={overflowItems}
+          ariaLabel={drawers.overflowAriaLabel}
+          customTriggerBuilder={(clickHandler, ref, isDisabled, isExpanded, ariaLabel) => (
+            <InternalButton
+              ref={ref}
+              ariaLabel={ariaLabel}
+              className={clsx(styles['drawers-trigger'], testutilStyles['drawers-trigger'])}
+              disabled={hasDrawerViewportOverlay}
+              formAction="none"
+              iconName="ellipsis"
+              badgeColor="red"
+              onClick={clickHandler}
+              variant="icon"
+              __nativeAttributes={{ 'aria-haspopup': true }}
+            />
+          )}
           onItemClick={({ detail }) => {
             handleDrawersClick(detail.id);
           }}
-          hasOverflowBadge={overflowItemHasBadge}
```

Comment:
Badges on button dropdown triggers are not supported in this PR. 

My suggestion would be to make this feature built into button dropdown, so it will automatically render the badge when any dropdown item has it. 

It is useful for our use-case, so can also be useful for customers with standalone use-case in button dropdown

Reply:
I see. So you're suggesting that we do that in a separate PR, and make it a built-in feature for the ButtonDropdown component. That's fine by me, but if there is design pushback we may have to re-implement it here this way.